login1/dbus.go

```diff
@@ -341,3 +364,35 @@ func (c *Conn) Subscribe(members ...string) chan *dbus.Signal {
 func (c *Conn) PowerOff(askForAuth bool) {
 	c.object.Call(dbusInterface+".PowerOff", 0, askForAuth)
 }
+
+func (c *Conn) getProperties(ctx context.Context, path dbus.ObjectPath, dbusInterface string) (map[string]dbus.Variant, error) {
+	if !path.IsValid() {
+		return nil, fmt.Errorf("invalid object path (%s)", path)
+	}
+
+	obj := c.conn.Object(dbusDest, path)
+
+	var props map[string]dbus.Variant
+	err := obj.CallWithContext(ctx, "org.freedesktop.DBus.Properties.GetAll", 0, dbusInterface).Store(&props)
+	if err != nil {
+		return nil, err
+	}
+
+	return props, nil
+}
+
+func (c *Conn) getProperty(ctx context.Context, path dbus.ObjectPath, dbusInterface, property string) (dbus.Variant, error) {
```

Comment:
I wonder if we should return the resulting `Variant` by reference instead, maybe?.